pkg/cloud/azure/actuators/clients.go

```diff
@@ -75,7 +75,7 @@ type AzureNetworkClient interface {
 	WaitForNetworkSGsCreateOrUpdateFuture(future network.SecurityGroupsCreateOrUpdateFuture) error
 
 	// Public Ip Address Operations
-	GetPublicIPAddress(resourceGroupName string, IPName string) (network.PublicIPAddress, error)
+	CreateOrUpdatePublicIPAddress(resourceGroupName string, IPName string) (network.PublicIPAddress, error)
```

Comment:
We don't need a `GetPublicAddress` method?

Reply:
^ Scratch that. Thoughts on renaming it to `EnsurePublicIPAddress`?

Reply:
I initially had this as `CreateOrGetPublicAddress` with separate logic for each, but we actually always want to reconcile it (update DNS name label, etc.), so I changed the method name.

I thought about `EnsurePublicIPAddress`, because it's similar to the usage in `cloud-provider-azure`, but decided on `CreateOrUpdate`, because it's more inline with the underlying Azure call (`s.scope.PublicIPAddresses.CreateOrUpdate`).